crates/viewer/re_viewer/src/app.rs

```diff
@@ -1950,6 +1951,20 @@ impl App {
         }
     }
 
+    /// Makes the first recording store active that is found for a given data source if any.
+    fn try_make_recording_from_source_active(
+        &self,
+        egui_ctx: &egui::Context,
+        store_hub: &mut StoreHub,
+        new_source: &SmartChannelSource,
+    ) {
+        if let Some(entity_db) = store_hub.find_recording_store_by_source(new_source) {
+            let store_id = entity_db.store_id().clone();
+            debug_assert!(store_id.is_recording()); // `find_recording_store_by_source` should have filtered for recordings rather than blueprints.
+            self.make_store_active_and_highlight(store_hub, egui_ctx, &store_id);
+        }
+    }
```

Comment:
Some version of that facility might be nice to have for the new recording panel impl 🤔

Reply:
happy to move around. was just a utility I've split after seeing the code for the third time